manifest-build-tools/application/reprove.py

```diff
@@ -195,6 +214,167 @@ def directory_for_repo(self, repo):
         repo_directory = os.path.join(self._builddir, repo_directory)
         return repo_directory
 
+    def branch_existing_repositories(self):
+        """
+        Issues create branch commands to repos in a provided manifest
+        :return: None
+        """
+        if self._branch_name is None:
+            print "Please provide the new branch name"
+            sys.exit(2)
+
+        repo_list = self._manifest.get_repositories()
+        if repo_list is None:
+            print "No repository list found in manifest file"
+            sys.exit(2)
+        else:
+            # Loop through list of repos and create specified branch on each
+            for repo in repo_list:
+                self.create_repo_branch(repo, self._branch_name)
+
+    def create_repo_branch(self, repo, branch):
+        """
+        create branch  on the repos in the manifest file
+        :param repo: A dictionary
+        :return: None
+        """
+        try:
+            repo_directory = self.directory_for_repo(repo)
```

Comment:
1. "checked-out-directory-name" is an optional field of repositories in manifest.
    It means the local directory name for check out repository.
    The default value is the name of repository. But we can specify it.
2.  Actually, the 'directory-name' is assigned in the function handle_manifest().
    The function create_repo_branch() may be called with a dictionary repo which has not the field 'directory-name'.